geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -30,32 +33,44 @@
     protected org.opengis.filter.Filter filter;
     private String filterString;
 
-    public JSimpleFeatureFilter(String sftString, String filterString) throws Exception {
-        this.sftString = sftString;
-        this.filterString = filterString;
-    }
+    private String transform;
+    private String transformSchema;
 
     private static abstract class AbstractFilter extends FilterBase {
         protected SimpleFeatureType sft;
         protected org.opengis.filter.Filter filter;
+        protected Boolean hasTransform;
+        protected KryoBufferSimpleFeature reusable;
         static Logger log = LoggerFactory.getLogger(JSimpleFeatureFilter.class);
 
-        AbstractFilter(String sftString, String filterString) {
+        AbstractFilter(String sftString, String filterString, String transform, String transformSchema) {
             log.debug("Instantiating new JSimpleFeatureFilter, sftString = {}, filterString = {}", sftString, filterString);
             sft = IteratorCache.sft(sftString);
             this.filter = IteratorCache.filter(sftString, filterString);
+
+            reusable = IteratorCache.serializer(sftString, SerializationOptions.withoutId()).getReusableFeature();
+            if(transform == null || transform.isEmpty() || transformSchema == null || transformSchema.isEmpty()) {
+                hasTransform = false;
+            } else {
+                hasTransform = true;
+                reusable.setTransforms(transform, SimpleFeatureTypes.createType("", transformSchema));
+            }
         }
 
 
         @Override
         public Cell transformCell(Cell v) throws IOException {
-            return v;
+            if(hasTransform) {
```

Comment:
Sorry just understood what he meant, just had to change that one line to "return v". Will push to new PR now